backend/src/akvo/lumen/lib/aggregation/bubble.clj

```diff
@@ -38,20 +38,23 @@
         column-label-name (get column-label "columnName")
         column-label-title (get column-label "title")
         column-bucket (utils/find-column columns (get query "bucketColumn"))
+        column-bucket-type (get column-bucket "type")
         column-bucket-name (get column-bucket "columnName")
         column-bucket-title (get column-bucket "title")
         max-points 2500
         have-aggregation (boolean column-bucket)
-        aggregation-method (get query "metricAggregation")
+        aggregation-method (if column-size
+                             (get query "metricAggregation")
+                             "count")
 
         sql-text-with-aggregation (str "SELECT "
-                                       (sql-aggregation-subquery aggregation-method "%1$s" column-size-type)
+                                       (sql-aggregation-subquery aggregation-method "%1$s" (or column-size-type column-bucket-type))
```

Comment:
@tangrammer does this line look correct?

Reply:
very handcraft but should work .... anyway we should review all aggregation namespaces trying to improve them